### DISCUSSION TITLE
How to wrap a div element on a table.

### DISCUSSION THREAD

**Discussion by YanceyOfficial** [markedjs/marked#3571] [Q&A] [answered]
I have a fixed width to render markdown, but the table is overflow.
<img width="765" alt="image" src="https://github.com/user-attachments/assets/013106f4-fcb7-4e3f-af2e-53d5c769a592" />

The simplest way is to wrap a div element on the table, and set the div to `overflow-x: scroll`. But I don't know how to do it.

```ts
renderer.table = ({ header, rows }: Tokens.Table) => {
      return `<div class="overflow-x-scroll">
                <table>
                    ???
                </table>
            </div>`
    }
```

<img width="679" alt="image" src="https://github.com/user-attachments/assets/55873ced-4a06-4ab2-bce7-39cc9c31a4aa" />

**Comment by UziTech** [comment, marked as answer]
You can use the original renderer to render the table.

```js
import { Marked, Renderer } from 'marked';

const renderer = new Renderer();

const marked = new Marked({
  renderer: {
    table(...arguments) {
      return `<div>${renderer.table.apply(this, arguments)}</div>`;
    },
  },
});

console.log(marked.parse(`
| a | b |
| - | - |
| c | d |
`));

```

**Reply by YanceyOfficial** [reply]
Thank you Tony! It works in my App.
